core/editor/src/document/document_message_handler.rs

```diff
@@ -566,33 +574,25 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 					responses.push_back(operation.into());
 				}
 			}
-			ReorderSelectedLayers(delta) => {
-				let selected_layer_paths: Vec<Vec<LayerId>> = self.selected_layers_sorted();
+			MoveSelectedLayersTo { path, insert_index } => {
+				responses.push_back(DocumentMessage::CopySelectedLayers.into());
+				responses.push_back(DocumentMessage::DeleteSelectedLayers.into());
+				responses.push_back(DocumentMessage::PasteLayers { path, insert_index }.into());
```

Comment:
What happens if one of the earlier operations fails? Nothing is preventing some of these operations from completing but not all, which results in a corrupt, or otherwise broken state for the user. I think I've already come across such a case.

Reply:
Good point, we will have to thing about that in general, we might need a notion of a transaction as it exists in database systems anyway